### ISSUE TITLE
Pi-hole: Initial setup help. Not reachable from other hosts.

### ISSUE THREAD

**Issue by redespace** [casperklein/homeassistant-addons#24]
I just installed this pihole addon on my HomeAssistant instance. What is not clear for me, is how to configure the IP of the PiHole instance. Is it running as "host" network mode ? If yes, that would mean that my HA and my PiHole should be on the same IP right?
In short --> What is the IP address that I need to set up on my Router as DNS server?

The addon started, and I can see the UI via the sidebar in HA, but not through the browser:

![image](https://github.com/user-attachments/assets/ab3e9357-90eb-48e9-a3c0-b9c309f72510)

![image](https://github.com/user-attachments/assets/d2eaec51-9765-42b1-af94-eeae5922466b)

Also, when trying doing DNS queries from my desktop PC using nslookup (nslookup facebook.com 192.168.2.62 {my HA IP} ), it is not hitting the PiHole instance --> There is no logs or query entries listed in the UI.

Also, regarding configuration --> Is there a way to change the Hostname and set a separate IP address (of course within the same subnet) of the PiHole instance?

**Comment by casperklein** [comment]
> Is it running as "host" network mode ?

No, the ports are mapped. But the outcome is the same in that case.

> that would mean that my HA and my PiHole should be on the same IP right?

Correct.

> What is the IP address that I need to set up on my Router as DNS server?

Your HA instance IP address.

> Is there a way to change the Hostname and set a separate IP address

Unfortunately that's not possible.

> when trying doing DNS queries from my desktop PC using nslookup (nslookup facebook.com 192.168.2.62 {my HA IP} ), it is not hitting the PiHole instance --> There is no logs or query entries listed in the UI.

I am not sure, why this is not working for you. I would check the HA supervisord logs for something related to that addon. Also check the output of `docker ps`. That should look similar to:

```
# docker ps -a

CONTAINER ID   IMAGE                                       COMMAND     CREATED        STATUS                  PORTS
d426cd917fbb   casperklein/homeassistant-pihole:2024.07.0  "/s6-init"  4 minutes ago  Up 4 minutes (healthy)  0.0.0.0:53->53/tcp, 0.0.0.0:53->53/udp, :::53->53/tcp, :::53->53/udp, 67/udp, 80/tcp
```

IPv4/6 tcp/udp port 53 is mapped to `0.0.0.0` (every interface),

**Comment by redespace** [comment]
Thank you Casper for your quick response.

I logged into my HA instance via SSH, and weirdly enough, it seems docker is not installed?

![image](https://github.com/user-attachments/assets/59a5bfd0-3386-4246-9673-cdd8d0c494ec)

How is this possible?

**Comment by redespace** [comment]
Done, I logged into 22222 and here is the output:

![image](https://github.com/user-attachments/assets/3e816de7-bdfa-48e3-956d-36ba8e81bfbe)

Seems that the container is running as expected.

Any ideas why I can't still reach the Pihole from the browser? :(
I tried http://192.168.2.62 and tried http://homeassistant.local/admin with no luck

**Comment by casperklein** [comment]
> Any ideas why I can't still reach the Pihole from the browser? :(

As seen in your screenshot, port 80 isn't exposed. You have to explicit enable it in the addon port configuration. However, you can access the pi-hole admin page via [HA ingress](https://www.home-assistant.io/blog/2019/04/15/hassio-ingress/) ("Show in sidebar" setting of the addon or the "OPEN WEB UI" button)

**Comment by redespace** [comment]
you are right Casper, I manually enabled port 80 and now it's exposed and I am able to see the UI through the browser. Nevertheless, I still don't see any queries and also nothing discovered by Pihole in the network:

![image](https://github.com/user-attachments/assets/c6dfd406-8bcb-40ff-adc1-2752c0a0708d)

**Comment by casperklein** [comment]
I have no idea, why port 80 is working and 53 not.

Verify that `homeassistant.local` is really resolving to `192.168.2.62`. Is `http://192.168.2.62/admin/`  working?

So from another host, `wget` is working, but `dig` not?

```bash
wget http://192.168.2.62/admin/
dig google.com @192.168.2.62
```

**Comment by casperklein** [comment]
@redespace Are there any updates on this?
